test/scenarios/update_file_via_tmp_file/scenario.js

```diff
@@ -0,0 +1,22 @@
+/* @flow */
+
+/*:: import type { Scenario } from '..' */
+
+module.exports = ({
+  side: 'local',
+  init: [{ ino: 1, path: 'file', content: 'initial content' }],
+  actions: [
+    { type: 'mv', src: 'file', dst: 'file.special-tmp' },
+    { type: 'create_file', path: 'file' },
+    { type: 'update_file', path: 'file', content: 'updated content #1' },
+    { type: 'delete', path: 'file.special-tmp' },
+    { type: 'update_file', path: 'file', content: 'updated content #2' }
```

Comment:
We should probably make sure we don't break a simple file replacement case, aka `deleted file + created file`.